Modules/mathmodule.c

```diff
@@ -2016,6 +2016,71 @@ math_factorial(PyObject *module, PyObject *arg)
 }
 
 
+/*[clinic input]
+math.lcm
+
+    x as a: object
+    y as b: object
+    /
+least common multiple of x and y
+[clinic start generated code]*/
+
+static PyObject *
+math_lcm_impl(PyObject *module, PyObject *a, PyObject *b)
+/*[clinic end generated code: output=6f83fb6d671074ba input=bd41b785dc2a4ff1]*/
+{
+    PyObject *g, *m, *f, *ab;
+    
+    a = PyNumber_Index(a);
+        if (a == NULL) {
+            return NULL;
+    }
+    b = PyNumber_Index(b);
+    if (b == NULL) {
+        return NULL;
+    }
+    if (a == 0 && b == 0) {
+        return 0;
+    }
+    else {
+        if (PyFloat_Check(a) || PyFloat_Check(b)) {
+            PyErr_Format(PyExc_TypeError,"'float' object cannot be an argument to lcm");
+            return 0;  
+        }
+        g = _PyLong_GCD(a, b);
+        Py_DECREF(a);
+        Py_DECREF(b);
+        Py_DECREF(b);
+        if (g == NULL) {
+            return NULL;
```

Comment:
If we get here then `_PyLong_GCD` failed. At this point, if you remove the incorrect DECREF's above, you still own references to both `a` and `b`. So in this branch you'd need to DECREF both `a` and `b`, and then return `NULL`.